### PR TITLE
Shorter literals

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: src/smt.rs [hunk 1]
```diff
@@ -36,8 +38,13 @@ pub struct SMTExpr {
 
 impl From<u64> for SMTExpr {
     fn from(input: u64) -> SMTExpr {
+        let formatted = if input < 1000 {
+            format!("(_ bv{input} 256)")
```
Comment (leonardoalt) [inline src/smt.rs:42]: Where's this one defined then?

File: tests/assert_pass/assignments_1.yul.smt2 [hunk 2]
```diff
@@ -19,28 +19,28 @@
 (declare-fun _keccak256_32_2066_0 ((_ BitVec 256)) (_ BitVec 256))
 (declare-fun _keccak256_64_2067_0 ((_ BitVec 256) (_ BitVec 256)) (_ BitVec 256))
 (declare-fun _keccak256_2068_0 ((Array (_ BitVec 256) (_ BitVec 8)) (_ BitVec 256) (_ BitVec 256)) (_ BitVec 256))
-(define-fun _stop_flag_2069_0 () (_ BitVec 256) #x0000000000000000000000000000000000000000000000000000000000000000)
-(define-fun _revert_flag_2070_0 () (_ BitVec 256) #x0000000000000000000000000000000000000000000000000000000000000000)
+(define-fun _stop_flag_2069_0 () (_ BitVec 256) (_ bv0 256))
```
Comment (leonardoalt) [inline tests/assert_pass/assignments_1.yul.smt2:22]: `bv0` seems undefined?

File: src/smt.rs [hunk 3]
```diff
@@ -360,7 +367,8 @@ pub fn as_const<V: Into<SMTExpr>>(sort: SMTSort, val: V) -> SMTExpr {
     }
 }
 
-pub fn literal(lit: String, sort: SMTSort) -> SMTExpr {
+fn literal(lit: String, sort: SMTSort) -> SMTExpr {
+    assert!(lit.starts_with("#x") || lit.starts_with("(_ bv") || &lit == "true" || &lit == "false");
```
Comment (leonardoalt) [inline src/smt.rs:371]: why would it start with `(_ bv`? this interface seems odd

File: tests/assert_pass/assignments_1.yul.smt2 (hunk 2 above)
Comment (chriseth) [reply tests/assert_pass/assignments_1.yul.smt2:22]: This is the weird SMT syntax to define bit vectors literals using decimal notation.

File: src/smt.rs (hunk 3 above)
Comment (chriseth) [reply src/smt.rs:371]: `#x...` is a bitvector literal in hex notation, `(_ bv... <bits>)` is a bitvector literal in decimal notation.

File: tests/assert_pass/assignments_1.yul.smt2 (hunk 2 above)
Comment (leonardoalt) [reply tests/assert_pass/assignments_1.yul.smt2:22]: lol

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**